### PR TITLE
Patch/correct update catalog functionality

### PR DESCRIPTION
**Pull request by tjellicoe-tpzuk**
Patch to correct update_catalog functionality to reindex items as well as collections when catalog ID is updated.
Also remove unnecessary parameter from discover-search function as part of general code clean-up.

### CODE REVIEW

File: stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py [hunk 1]
```diff
@@ -1922,7 +1957,7 @@ async def execute_discovery_search(
         try:
             es_response = await search_task
         except exceptions.NotFoundError:
-            raise NotFoundError(f"Catalogs '{catalog_ids}' do not exist")
+            raise NotFoundError("Catalog or Collection missing during search.")
```
Comment (hcollingwood) [inline stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py]: Is it worth including which Catalog(s)/Collection(s) haven't been found?

Comment (tjellicoe-tpzuk) [reply stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py]: As a matter of fact, I don't think this error will ever be triggered here as the indices we are searching are only prefixes which are created at boot-time, therefore they will never be missing. I will update this to remove the unnecessary try-except block.

File: stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py [hunk 2]
```diff
@@ -1933,8 +1928,6 @@ async def execute_discovery_search(
 
         query = search.query.to_dict() if search.query else None
 
-        index_param = "document"  # indices(collection_ids)
```
Comment (hcollingwood) [inline stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py:1901]: document isn't needed anymore?

**Review by hcollingwood — APPROVED** [review]

File: stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py (hunk 2 above)
Comment (tjellicoe-tpzuk) [reply stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/database_logic.py:1901]: Yes, I believe this was included as a placeholder, but we can instead directly code the index into the function calls on line #1944 and #1237 respectively.

**Pull request merged**